packages/framework/src/hooks/useStyles.ts

```diff
@@ -0,0 +1,81 @@
+import { useAtomValue } from "jotai";
+import { type CSSProperties, useMemo } from "react";
+import {
+  compact,
+  get,
+  intersection,
+  isEmpty,
+  isString,
+  keys,
+  merge,
+  omit,
+} from "lodash-es";
+import { type EnsembleThemeModel } from "../shared";
+import { themeAtom } from "../state";
+import { useEvaluate } from "./useEvaluate";
+
+const resolveStyleNames = (
+  styleNames: string[],
+  theme: EnsembleThemeModel,
+): CSSProperties => {
+  const appStyleNames = keys(theme.Styles);
+  const overlappingNames = intersection(styleNames, appStyleNames);
+  if (isEmpty(overlappingNames)) {
+    return {};
+  }
+
+  const appliedStyles: CSSProperties = {};
+  overlappingNames.forEach((name) => {
+    merge(appliedStyles, get(theme.Styles, name));
+  });
+
+  return appliedStyles;
+};
+
+export const useStyles = <T extends { [key: string]: unknown }>(
+  values: T,
+): CSSProperties | undefined => {
+  const namedStyles = get(values, ["styles", "names"]) as unknown;
+  const classStyles = get(values, ["styles", "className"]) as unknown;
+
+  const themeContext = useAtomValue(themeAtom);
+
+  const { classStylesArray, nameStylesArray } = useMemo(() => {
+    const stringSplitter = (styles: string): string[] =>
+      // eslint-disable-next-line prefer-named-capture-group
+      styles.split(/([a-zA-Z0-9_-]+)\s+(\${.+?})\s+(\w+)/).filter(Boolean);
+    return {
+      classStylesArray: isString(classStyles)
+        ? stringSplitter(classStyles)
+        : [],
+      nameStylesArray: isString(namedStyles) ? stringSplitter(namedStyles) : [],
```

Comment:
Does this still work if there are mixed static and expressions, for example

```yaml
styles:
  className: "myStyle ${ensemble.storage.get('myOtherStyle')}"
```

I think a better way to do this is to just treat the entire thing as a string interpolation.

Reply:
Yes, and we just can't treat this as entire string interpolation, as due to classname we need to add '.' as prefix on each classname